lib/Parse/ParseStmt.cpp

```diff
@@ -2476,6 +2476,20 @@ ParserResult<CaseStmt> Parser::parseStmtCase(bool IsActive) {
 
   assert(!CaseLabelItems.empty() && "did not parse any labels?!");
 
+  // Add a scope so that the parser can find our body bound decls if it emits
+  // optimized accesses.
+  Optional<Scope> BodyScope;
+  if (CaseBodyDecls) {
+    BodyScope.emplace(this, ScopeKind::CaseVars);
+    for (auto *v : *CaseBodyDecls) {
+      setLocalDiscriminator(v);
+      // If we had any bad redefinitions, we already diagnosed them against the
+      // first case label item.
+      if (v->hasName())
+        addToScope(v, false /*diagnoseRedefinitions*/);
```

Comment:
Is this always going to be a redefinition?

Reply:
Yes.

Reply:
Maybe add a new entry point, like addRedefinitionToScope(), that changes the logic around and asserts if its not a redefinition.

Reply:
SGTM

Reply:
Actually that is not what we want. We actually just do not want to emit
the error here. The reason why is that we are not always going to
have a redefinition But in the case where we do have the
redefinition, we only want to emit a single error (the one from the case label). That is why for
the case body decls I have wired through the flag to ignore the error /if/ it occurs.

Reply:
@slavapestov ping?